src/CppUTestExt/MockFailure.cpp

```diff
@@ -43,7 +43,7 @@ class MockFailureReporterTestTerminator : public NormalTestTerminator
         if (crashOnFailure_)
```

Comment:
I thought we were going to do the mock after this PR. If not, then this probably need to change as this functionality is not moved to CppUMock.

Reply:
We are. This change (to call the current TestTerminator, rather than a hard coded one) is to support case #3 - when the command-line option is provided, all failures will cause a crash.

Reply:
Ok, but if we do the mock in a separate PR, then I suggest we do that.

If we do the mock in this PR, then we need to remove the crashOnFailure_ boolean out of the mock and instead use the CppUTest one. Now it is just a half done implementation.

Reply:
I do not agree that it is half done. This implements precisely what was agreed for the initial feature, in terms of the 3 scenarios. It does mean that we have two related mechanisms in mock that we should look at harmonizing later but this change is the simplest change that will cause the 3 tests for the 3 scenarios to pass.

Reply:
Sure... but it adds functionality duplication. I would prefer to either not add this functionality or do it proper. Leaving code like this makes it easy to not do the cleaning up... which leads to hard to maintain legacy code.